Mapping/Loader/YamlLoader.php

```diff
@@ -0,0 +1,167 @@
+    {
+        $classMetadata = [];
+
+        $classMetadata['perEnvironment'] = true;
+        if (isset($config['perEnvironment'])) {
+            $classMetadata['perEnvironment'] = $config['perEnvironment'];
+        }
+
+        $classMetadata['autoIndex'] = true;
+        if (isset($config['autoIndex'])) {
+            $classMetadata['autoIndex'] = $config['autoIndex'];
+        }
+
+        if (isset($config['algoliaName'])) {
+            $classMetadata['algoliaName'] = $config['algoliaName'];
+        }
+
+        foreach (Index::$algoliaSettingsProps as $field) {
+            if (isset($config[$field])) {
+                $classMetadata[$field] = $config[$field];
+            }
+        }
+
+        $index = new Index();
+        $index->setAlgoliaNameFromClass($className);
+        $index->updateSettingsFromArray($classMetadata);
+        $description->setIndex($index);
+    }
+
+    /**
+     * @param \ReflectionClass $class
+     * @param $className
+     * @param $config
+     * @param $description
+     */
+    protected function extractPropertyMetadata(\ReflectionClass $class, $className, $config, Description $description)
+    {
+        foreach ($class->getProperties() as $property) {
+            if ($className !== $property->class) {
+                continue;
+            }
+
+            $propertyName = $property->getName();
+
+            if (isset($config['properties'][$propertyName])) {
+                $propertyConfig = $config['properties'][$propertyName];
+
+                if (!isset($propertyConfig['type']) && !is_scalar($propertyConfig['type'])) {
+                    throw new \RuntimeException('The "type" attribute must be set for each property.');
+                }
+
+                if ((string) $propertyConfig['type'] === 'attribute') {
+                    $field = new Property();
+                    $field->setName($property->getName());
+
+                    if (isset($propertyConfig['algoliaName'])) {
+                        $field->setAlgoliaName($propertyConfig['algoliaName']);
+                    } else {
+                        $field->setAlgoliaName($field->getName());
+                    }
+
+                    $description->addProperty($field);
+                } elseif ((string) $propertyConfig['type'] === 'id') {
+                    $description->addIdentifierAttributeName($propertyName);
+                } else {
+                    throw new \RuntimeException('The "type" attribute is wrongly configured.');
+                }
+            }
+        }
+    }
+
+    /**
+     * @param \ReflectionClass $class
+     * @param $className
+     * @param $config
+     * @param $description
+     */
+    protected function extractMethodMetadata(\ReflectionClass $class, $className, $config, Description $description)
+    {
+        foreach ($class->getMethods() as $method) {
+            if ($className !== $method->class) {
```

Comment:
The same as https://github.com/algolia/AlgoliaSearchBundle/pull/53/files#r91077073 but related to inherited methods.

What is the intention of skipping inherited properties? This requires to redefine all properties to be indexed in an extended class. This is especially tricky if a project depends f.e. on entities from a third party project. At the moment I can see only a lot of not required boilerplate code coming up due to this restriction...?!